src/node_i18n.cc

```diff
@@ -221,26 +218,18 @@ MaybeLocal<Object> TranscodeUcs2FromUtf8(Environment* env,
                                          const char* source,
                                          const size_t source_length,
                                          UErrorCode* status) {
-  *status = U_ZERO_ERROR;
-  MaybeStackBuffer<UChar> destbuf;
-  int32_t result_length;
-  u_strFromUTF8(*destbuf, destbuf.capacity(), &result_length,
-                source, source_length, status);
-  MaybeLocal<Object> ret;
-  if (U_SUCCESS(*status)) {
-    destbuf.SetLength(result_length);
-    ret = ToBufferEndian(env, &destbuf);
-  } else if (*status == U_BUFFER_OVERFLOW_ERROR) {
-    *status = U_ZERO_ERROR;
-    destbuf.AllocateSufficientStorage(result_length);
-    u_strFromUTF8(*destbuf, result_length, &result_length,
-                  source, source_length, status);
-    if (U_SUCCESS(*status)) {
-      destbuf.SetLength(result_length);
-      ret = ToBufferEndian(env, &destbuf);
-    }
+  size_t expected_utf16_length =
+      simdutf::utf16_length_from_utf8(source, source_length);
+  MaybeStackBuffer<UChar> destbuf(expected_utf16_length);
+  auto actual_length =
+      simdutf::convert_utf8_to_utf16(source, source_length, destbuf.out());
+
+  if (actual_length == 0) {
+    *status = U_INVALID_CHAR_FOUND;
+    return {};
   }
-  return ret;
+
+  return ToBufferEndian(env, &destbuf);
```

Comment:
It looks like `simdutf::convert_utf8_to_utf16` already accounts for endianness, so the `ToBufferEndian()` call is not needed?
Refs: https://github.com/simdutf/simdutf/blob/f058d353140f27257d68a5f45ab9ce35b2ed756c/src/implementation.cpp#L946-L951